iib/workers/tasks/build_add_deprecations.py

```diff
@@ -0,0 +1,198 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import os
+import json
+import logging
+import tempfile
+from typing import Dict, Optional, Set
+
+from iib.common.common_utils import get_binary_versions
+from iib.common.tracing import instrument_tracing
+from iib.workers.api_utils import set_request_state
+from iib.workers.tasks.build import (
+    _add_label_to_index,
+    _build_image,
+    _cleanup,
+    _create_and_push_manifest_list,
+    _push_image,
+    _update_index_image_build_state,
+    _update_index_image_pull_spec,
+)
+from iib.workers.tasks.celery import app
+from iib.workers.tasks.fbc_utils import get_catalog_dir
+from iib.workers.tasks.opm_operations import (
+    Opm,
+    generate_cache_locally,
+    opm_generate_dockerfile,
+    opm_validate,
+    verify_operators_exists,
+)
+from iib.workers.tasks.utils import (
+    prepare_request_for_build,
+    request_logger,
+    RequestConfigAddDeprecations,
+    IIBError,
+    get_worker_config,
+)
+
+__all__ = ['handle_add_deprecations_request']
+
+log = logging.getLogger(__name__)
+
+
+@app.task
+@request_logger
+@instrument_tracing(
+    span_name="workers.tasks.build.handle_add_deprecations_request",
+    attributes=get_binary_versions(),
+)
+def handle_add_deprecations_request(
+    request_id: int,
+    operator_package: str,
+    deprecation_schema: str,
+    from_index: Optional[str] = None,
+    binary_image: Optional[str] = None,
+    overwrite_from_index: bool = False,
```

Comment:
Should this be `overwrite_from_index: Optional[bool] = False`? If not, this should probably be moved up before all the Optional params begin